// @azure/cognitiveservices-imagesearch/esm/models/parameters.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for
 * license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is
 * regenerated.
 */
import * as msRest from "@azure/ms-rest-js";
export var acceptLanguage = {
    parameterPath: [
        "options",
        "acceptLanguage"
    ],
    mapper: {
        serializedName: "Accept-Language",
        type: {
            name: "String"
        }
    }
};
export var aspect = {
    parameterPath: [
        "options",
        "aspect"
    ],
    mapper: {
        serializedName: "aspect",
        type: {
            name: "String"
        }
    }
};
export var clientId = {
    parameterPath: [
        "options",
        "clientId"
    ],
    mapper: {
        serializedName: "X-MSEdge-ClientID",
        type: {
            name: "String"
        }
    }
};
export var clientIp = {
    parameterPath: [
        "options",
        "clientIp"
    ],
    mapper: {
        serializedName: "X-MSEdge-ClientIP",
        type: {
            name: "String"
        }
    }
};
export var color = {
    parameterPath: [
        "options",
        "color"
    ],
    mapper: {
        serializedName: "color",
        type: {
            name: "String"
        }
    }
};
export var contentType = {
    parameterPath: [
        "options",
        "contentType"
    ],
    mapper: {
        serializedName: "Content-Type",
        type: {
            name: "String"
        }
    }
};
export var count = {
    parameterPath: [
        "options",
        "count"
    ],
    mapper: {
        serializedName: "count",
        type: {
            name: "Number"
        }
    }
};
export var countryCode = {
    parameterPath: [
        "options",
        "countryCode"
    ],
    mapper: {
        serializedName: "cc",
        type: {
            name: "String"
        }
    }
};
export var cropBottom = {
    parameterPath: [
        "options",
        "cropBottom"
    ],
    mapper: {
        serializedName: "cab",
        type: {
            name: "Number"
        }
    }
};
export var cropLeft = {
    parameterPath: [
        "options",
        "cropLeft"
    ],
    mapper: {
        serializedName: "cal",
        type: {
            name: "Number"
        }
    }
};
export var cropRight = {
    parameterPath: [
        "options",
        "cropRight"
    ],
    mapper: {
        serializedName: "car",
        type: {
            name: "Number"
        }
    }
};
export var cropTop = {
    parameterPath: [
        "options",
        "cropTop"
    ],
    mapper: {
        serializedName: "cat",
        type: {
            name: "Number"
        }
    }
};
export var cropType = {
    parameterPath: [
        "options",
        "cropType"
    ],
    mapper: {
        serializedName: "ct",
        type: {
            name: "String"
        }
    }
};
export var endpoint = {
    parameterPath: "endpoint",
    mapper: {
        required: true,
        serializedName: "Endpoint",
        defaultValue: 'https://api.cognitive.microsoft.com',
        type: {
            name: "String"
        }
    },
    skipEncoding: true
};
export var freshness = {
    parameterPath: [
        "options",
        "freshness"
    ],
    mapper: {
        serializedName: "freshness",
        type: {
            name: "String"
        }
    }
};
export var height = {
    parameterPath: [
        "options",
        "height"
    ],
    mapper: {
        serializedName: "height",
        type: {
            name: "Number"
        }
    }
};
export var id = {
    parameterPath: [
        "options",
        "id"
    ],
    mapper: {
        serializedName: "id",
        type: {
            name: "String"
        }
    }
};
export var imageContent = {
    parameterPath: [
        "options",
        "imageContent"
    ],
    mapper: {
        serializedName: "imageContent",
        type: {
            name: "String"
        }
    }
};
export var imageType = {
    parameterPath: [
        "options",
        "imageType"
    ],
    mapper: {
        serializedName: "imageType",
        type: {
            name: "String"
        }
    }
};
export var imageUrl = {
    parameterPath: [
        "options",
        "imageUrl"
    ],
    mapper: {
        serializedName: "imgUrl",
        type: {
            name: "String"
        }
    }
};
export var insightsToken = {
    parameterPath: [
        "options",
        "insightsToken"
    ],
    mapper: {
        serializedName: "insightsToken",
        type: {
            name: "String"
        }
    }
};
export var license = {
    parameterPath: [
        "options",
        "license"
    ],
    mapper: {
        serializedName: "license",
        type: {
            name: "String"
        }
    }
};
export var location = {
    parameterPath: [
        "options",
        "location"
    ],
    mapper: {
        serializedName: "X-Search-Location",
        type: {
            name: "String"
        }
    }
};
export var market = {
    parameterPath: [
        "options",
        "market"
    ],
    mapper: {
        serializedName: "mkt",
        type: {
            name: "String"
        }
    }
};
export var maxFileSize = {
    parameterPath: [
        "options",
        "maxFileSize"
    ],
    mapper: {
        serializedName: "maxFileSize",
        type: {
            name: "Number"
        }
    }
};
export var maxHeight = {
    parameterPath: [
        "options",
        "maxHeight"
    ],
    mapper: {
        serializedName: "maxHeight",
        type: {
            name: "Number"
        }
    }
};
export var maxWidth = {
    parameterPath: [
        "options",
        "maxWidth"
    ],
    mapper: {
        serializedName: "maxWidth",
        type: {
            name: "Number"
        }
    }
};
export var minFileSize = {
    parameterPath: [
        "options",
        "minFileSize"
    ],
    mapper: {
        serializedName: "minFileSize",
        type: {
            name: "Number"
        }
    }
};
export var minHeight = {
    parameterPath: [
        "options",
        "minHeight"
    ],
    mapper: {
        serializedName: "minHeight",
        type: {
            name: "Number"
        }
    }
};
export var minWidth = {
    parameterPath: [
        "options",
        "minWidth"
    ],
    mapper: {
        serializedName: "minWidth",
        type: {
            name: "Number"
        }
    }
};
export var modules = {
    parameterPath: [
        "options",
        "modules"
    ],
    mapper: {
        serializedName: "modules",
        type: {
            name: "Sequence",
            element: {
                type: {
                    name: "String"
                }
            }
        }
    },
    collectionFormat: msRest.QueryCollectionFormat.Csv
};
export var offset = {
    parameterPath: [
        "options",
        "offset"
    ],
    mapper: {
        serializedName: "offset",
        type: {
            name: "Number"
        }
    }
};
export var query = {
    parameterPath: "query",
    mapper: {
        required: true,
        serializedName: "q",
        type: {
            name: "String"
        }
    }
};
export var safeSearch = {
    parameterPath: [
        "options",
        "safeSearch"
    ],
    mapper: {
        serializedName: "safeSearch",
        type: {
            name: "String"
        }
    }
};
export var setLang = {
    parameterPath: [
        "options",
        "setLang"
    ],
    mapper: {
        serializedName: "setLang",
        type: {
            name: "String"
        }
    }
};
export var size = {
    parameterPath: [
        "options",
        "size"
    ],
    mapper: {
        serializedName: "size",
        type: {
            name: "String"
        }
    }
};
export var userAgent = {
    parameterPath: [
        "options",
        "userAgent"
    ],
    mapper: {
        serializedName: "User-Agent",
        type: {
            name: "String"
        }
    }
};
export var width = {
    parameterPath: [
        "options",
        "width"
    ],
    mapper: {
        serializedName: "width",
        type: {
            name: "Number"
        }
    }
};
export var xBingApisSDK = {
    parameterPath: "xBingApisSDK",
    mapper: {
        required: true,
        isConstant: true,
        serializedName: "X-BingApis-SDK",
        defaultValue: 'true',
        type: {
            name: "String"
        }
    }
};
